File: CoreJava/Tdd-app/src/com/techlabs/calculator/Calculator.java
package com.techlabs.calculator;

public class Calculator {
	public int squareOdd(int number) throws Exception {
		if (number % 2 == 0) {
			throw new RuntimeException("Number should not be even");
		} else if (number < 0) {
			throw new RuntimeException("Number should be greater than zero");
		} else {
			return number * number;
		}
	}
}

File: DotNET/DLL/ClassLibrarySolution/MathComponentLib/Calculator.cs
using System;

namespace MathComponentLib
{
    public class Calculator
    {
        public int CubeEvenNum(int num)
        {
            if (IsEvenNum(num))
                return num * num * num;
            else
                throw new Exception("Number is not Even");
        }

        private bool IsEvenNum(int num)
        {
            if (num % 2 == 0)
                return true;
            else
                return false;
        }
    }
}
